modules/core/src/main/java/org/apache/ignite/internal/processors/rest/GridRestProcessor.java

```diff
@@ -245,19 +245,19 @@
         boolean authenticationEnabled = ctx.authentication().enabled();
         boolean securityEnabled = ctx.security().enabled();
 
-        if (authenticationEnabled || securityEnabled) {
-            Session ses;
+        Session ses;
 
-            try {
-                ses = session(req);
-            }
-            catch (IgniteAuthenticationException e) {
-                return new GridFinishedFuture<>(new GridRestResponse(STATUS_AUTH_FAILED, e.getMessage()));
-            }
-            catch (IgniteCheckedException e) {
-                return new GridFinishedFuture<>(new GridRestResponse(STATUS_FAILED, e.getMessage()));
-            }
+        try {
+            ses = session(req);
```

Comment:
Using such an approach you will lose user attributes when session will expire.

Reply:
Moved saving attributes to GridNioSession.